junit/kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/crud/PatchHandler.java

```diff
@@ -63,6 +63,7 @@ public MockResponse handle(String path, String contentType, String requestBody)
       fullPatch = persistence.merge(currentResource, requestBody);
     }
     validatePath(query, fullPatch);
+    validateResourceVersion(currentResource, fullPatch);
```

Comment:
We're validating the resource version in all patch scenarios which AFAIU is not correct

Reply:
This validation would fail only if the user includes a resourceVersion in the patch.  I believe this is correct.